keel-core/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactVersionLinks.kt

```diff
@@ -51,22 +52,40 @@ class ArtifactVersionLinks(private val scmInfo: ScmInfo, private val cacheFactor
     } else null
   }
 
+  fun getScmType(commitLink: String): String? {
+    val commitURL = URL(commitLink)
```

Comment:
Do we need to worry about the potential exception being thrown here?

Reply:
Yes good call. I didn't like that. I fixed this by using getOrNull.

Reply:
I was talking about the`java.net.URL` constructor there. It throws `MalformedURLException`. So if given commit link is somehow malformed or empty, the exception will bubble up. `newerGitMetadata?.commitInfo?.link?` will check if it's not null, but not empty / blank string.